portfolio/src/main/webapp/assets/js/testimonials.js

```diff
@@ -110,4 +110,104 @@ function renderVoiceButton(){
 function onPageLoad(){
     getTestimonials();
     initializeVoiceControl();
-}
\ No newline at end of file
+}
+
+/** Takes two words and builds/returns the starting DP table to work with */
+function buildInitialTable(w1, w2){
+    // create 2d matrix dp table using array of arrays
+    dp = [];
+    
+    // create first row representing edit distance of empty vs w2
+    r1 = []
+    for(i=0; i<w2.length+1; i++){
+      r1.push(i);
+    }
+    
+    // add first row to dp table
+    dp.push(r1);
+
+    // Fill in remaining rows TODO: optimize to O(N) rather than O(N^2)
+    for(i=1; i < w1.length+1; i++){
+        row = [i];
+        for(j=0; j<w2.length; j++){
+          row.push(null);
+        }
+        dp.push(row);
+    }
+    return dp;
+}
+
+function findEditDistance(w1,w2){
```

Comment:
When you using DP it is super important to add a comment describing exactly what its computing since its hard for a reviewer to understand it without it. Ideally you would also include the recurrence in the comment.